src/Sign.Core/CertificateServices/CertificateStoreService.cs

```diff
@@ -147,17 +150,17 @@ private Task<X509Certificate2> GetStoreCertificateAsync()
             throw new ArgumentException(_isPrivateMachineKeyContainer ? Resources.CertificateNotFoundInMachineStore : Resources.CertificateNotFoundInUserStore);
         }
 
-        private static bool TryFindCertificate(StoreLocation storeLocation, string sha1Fingerprint, [NotNullWhen(true)] out X509Certificate2? certificate)
+        private bool TryFindCertificate(StoreLocation storeLocation, string certificateFingerprint, [NotNullWhen(true)] out X509Certificate2? certificate)
         {
             // Check machine certificate store.
             using (var store = new X509Store(StoreName.My, storeLocation))
             {
                 store.Open(OpenFlags.OpenExistingOnly | OpenFlags.ReadOnly);
-                var certificates = store.Certificates.Find(X509FindType.FindByThumbprint, sha1Fingerprint, validOnly: false);
+                var certificates = store.Certificates.Find(X509FindType.FindByThumbprint, certificateFingerprint, validOnly: false);
```

Comment:
Have you verified that this works?  You should have tests for it.

Looking at [.NET code](https://github.com/dotnet/runtime/blob/736ca4bede96c2cc98a13da8f8dfd6bb8688f15d/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/ManagedCertificateFinder.cs#L40-L63), it looks like `FindByThumbprint` only searches by SHA-1 fingerprint, so passing anything else would just fail.